analytics_data_api/utils.py

```diff
@@ -84,3 +94,92 @@ def date_range(start_date, end_date, delta=datetime.timedelta(days=1)):
     while cur_date < end_date:
         yield cur_date
         cur_date += delta
+
+
+def get_course_report_download_details(course_id, report_name):
+    """
+    Determine the path that the report file should be located at,
+    then return metadata sufficient for downloading it.
+    """
+    report_location_template = getattr(settings, 'COURSE_REPORT_FILE_LOCATION_TEMPLATE')
+    report_location = report_location_template.format(
+        course_id=course_id,
+        report_name=report_name
+    )
+    if getattr(settings, 'DEFAULT_FILE_STORAGE', None) != AWS_S3_FILE_STORAGE_CLASS:
+        # At present, we do not support linking from storage other than S3.
+        raise CannotCreateReportDownloadLinkError
+
+    if not default_storage.exists(report_location):
```

Comment:
However, I do agree that it's important to check that django.storage is configured and available.  You can do this by replacing this stanza with:
```
    try:
        if not default_storage.exists(report_location):
            raise ReportFileNotFoundError(course_id=course_id, report_name=report_name)
    except (NotImplementedError, ImportError, SuspiciousFileOperation):
        raise CannotCreateReportDownloadLinkError
```

* `NotImplementedError` catches the case where there's a django.storage chosen that doesn't implement `exists()`
* `ImportError` catches the case where there's an invalid django.storage class configured
* `SuspiciousFileOperation` catches the case where no settings.DEFAULT_FILE_STORAGE is provided ([see above](#r73460844)), and so django.storage is using the default FileSystemStorage, and our report file sit outside the settings.MEDIA_DIR.